### PR TITLE
Refactor FXIOS-10205 [Swiftlint] Resolve 1 implicitly_unwrapped_optional violations in LegacyFeatureFlagsManager

### PR DESCRIPTION
**Pull request by tonell-m**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10205)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22337)

## :bulb: Description

Continuing resolving `implicitly_unwrapped_optional` violations in order to enable the rule for further developments.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift [hunk 1]
```diff
@@ -51,6 +51,10 @@ class LegacyFeatureFlagsManager: HasNimbusFeatureFlags {
     public func isFeatureEnabled(_ featureID: NimbusFeatureFlagID,
                                  checking channelsToCheck: FlaggableFeatureCheckOptions
     ) -> Bool {
+        guard let profile else {
+            return false
+        }
```
Comment (MattLichtenstein) [inline firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift:56]:
```suggestion
        guard let profile else { return false }
```

File: firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift [hunk 2]
```diff
@@ -78,6 +82,10 @@ class LegacyFeatureFlagsManager: HasNimbusFeatureFlags {
     /// binary state. Further information on return types can be found in
     /// `FlaggableFeatureOptions`
     public func getCustomState<T>(for featureID: NimbusFeatureFlagWithCustomOptionsID) -> T? {
+        guard let profile else {
+            return nil
+        }
```
Comment (MattLichtenstein) [inline firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift:87]:
```suggestion
        guard let profile else { return nil }
```

File: firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift [hunk 3]
```diff
@@ -101,6 +109,10 @@ class LegacyFeatureFlagsManager: HasNimbusFeatureFlags {
 
     /// Set a feature that has a binary state to on or off
     public func set(feature featureID: NimbusFeatureFlagID, to desiredState: Bool, isDebug: Bool = false) {
+        guard let profile else {
+            return
+        }
```
Comment (MattLichtenstein) [inline firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift:114]:
```suggestion
        guard let profile else { return }
```

File: firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift [hunk 4]
```diff
@@ -119,6 +131,10 @@ class LegacyFeatureFlagsManager: HasNimbusFeatureFlags {
         feature featureID: NimbusFeatureFlagWithCustomOptionsID,
         to desiredState: T
     ) {
+        guard let profile else {
+            return
+        }
```
Comment (MattLichtenstein) [inline firefox-ios/Client/FeatureFlags/LegacyFeatureFlagsManager.swift:136]:
```suggestion
        guard let profile else { return }
```

**Review by MattLichtenstein — CHANGES_REQUESTED** [review]
Small spacing changes, but otherwise looks good!

**Comment by tonell-m** [comment]
@MattLichtenstein Done! Thanks for the review

**Review by MattLichtenstein — APPROVED** [review]

**Pull request merged**